diff --git a/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm b/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm
index f1d55dc..fdd59d2 100644
--- a/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm
+++ b/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm
@@ -42,29 +42,33 @@ using namespace facebook::react;
   return self;
 }
 
-#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
-    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
-
 - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
 {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   const auto &oldViewProps = *std::static_pointer_cast<LiquidGlassContainerViewProps const>(_props);
   const auto &newViewProps = *std::static_pointer_cast<LiquidGlassContainerViewProps const>(props);
   
   if (oldViewProps.spacing != newViewProps.spacing) {
     [_view setSpacing:newViewProps.spacing];
   }
+#endif
 
   [super updateProps:props oldProps:oldProps];
 }
 
 - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   [_view.contentView insertSubview:childComponentView atIndex:index];
+#else
+  [_view insertSubview:childComponentView atIndex:index];
+#endif
 }
 
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
   [childComponentView removeFromSuperview];
 }
-#endif
 
 @end
 
diff --git a/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm b/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm
index 2cfc99b..e59f43b 100644
--- a/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm
+++ b/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm
@@ -45,16 +45,19 @@ using namespace facebook::react;
   return self;
 }
 
-#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
-    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
 - (void)layoutSubviews {
   [super layoutSubviews];
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   _view.layer.cornerRadius = self.layer.cornerRadius;
   _view.layer.cornerCurve = self.layer.cornerCurve;
+#endif
 }
 
 - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
 {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   const auto &oldViewProps = *std::static_pointer_cast<LiquidGlassViewProps const>(_props);
   const auto &newViewProps = *std::static_pointer_cast<LiquidGlassViewProps const>(props);
   BOOL needsSetup = NO;
@@ -125,6 +128,7 @@ using namespace facebook::react;
   if (needsSetup) {
     [_view setupView];
   }
+#endif
   
   [super updateProps:props oldProps:oldProps];
 }
@@ -138,6 +142,8 @@ using namespace facebook::react;
   
   _needsInvalidateLayer = NO;
   
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   if (@available(iOS 26.0, tvOS 26.0, *)) {
     const auto borderMetrics = _props->resolveBorderMetrics(_layoutMetrics);
 
@@ -156,6 +162,7 @@ using namespace facebook::react;
 
     _view.cornerConfiguration = [UICornerConfiguration configurationWithTopLeftRadius:topLeft topRightRadius:topRight bottomLeftRadius:bottomLeft bottomRightRadius:bottomRight];
   }
+#endif
 }
 
 - (void)updateLayoutMetrics:(const LayoutMetrics &)layoutMetrics
@@ -170,13 +177,16 @@ using namespace facebook::react;
 }
 
 - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   [_view.contentView insertSubview:childComponentView atIndex:index];
+#else
+  [_view insertSubview:childComponentView atIndex:index];
+#endif
 }
 
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
   [childComponentView removeFromSuperview];
 }
 
-#endif
-
 @end
diff --git a/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewContainerNativeComponent.ts b/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewContainerNativeComponent.ts
index 40dae13..f074a4b 100644
--- a/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewContainerNativeComponent.ts
+++ b/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewContainerNativeComponent.ts
@@ -1,15 +1,15 @@
 import {
   codegenNativeComponent,
   type ViewProps,
-  type CodegenTypes,
 } from 'react-native';
+import type { Float } from 'react-native/Libraries/Types/CodegenTypes';
 
 export interface NativeProps extends ViewProps {
   /**
    * The distance between elements at which they begin to merge.
    * Defaults to 0.
    */
-  spacing?: CodegenTypes.Float;
+  spacing?: Float;
 }
 
 export default codegenNativeComponent<NativeProps>('LiquidGlassContainerView');
diff --git a/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewNativeComponent.ts b/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewNativeComponent.ts
index 4e87aae..fa4e8bc 100644
--- a/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewNativeComponent.ts
+++ b/node_modules/@callstack/liquid-glass/lib/module/LiquidGlassViewNativeComponent.ts
@@ -2,17 +2,17 @@ import {
   codegenNativeComponent,
   type ViewProps,
   type ColorValue,
-  type CodegenTypes,
 } from 'react-native';
+import type { Float, WithDefault } from 'react-native/Libraries/Types/CodegenTypes';
 
 export interface NativeProps extends ViewProps {
   interactive?: boolean;
-  effect?: CodegenTypes.WithDefault<'clear' | 'regular' | 'none', 'regular'>;
-  animated?: CodegenTypes.WithDefault<boolean, true>;
-  animationDuration?: CodegenTypes.WithDefault<CodegenTypes.Float, 0>;
-  hasAnimationDuration?: CodegenTypes.WithDefault<boolean, false>;
+  effect?: WithDefault<'clear' | 'regular' | 'none', 'regular'>;
+  animated?: WithDefault<boolean, true>;
+  animationDuration?: WithDefault<Float, 0>;
+  hasAnimationDuration?: WithDefault<boolean, false>;
   tintColor?: ColorValue;
-  colorScheme?: CodegenTypes.WithDefault<'light' | 'dark' | 'system', 'system'>;
+  colorScheme?: WithDefault<'light' | 'dark' | 'system', 'system'>;
 }
 
 export default codegenNativeComponent<NativeProps>('LiquidGlassView');
diff --git a/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewContainerNativeComponent.d.ts b/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewContainerNativeComponent.d.ts
index f96c2e2..4149028 100644
--- a/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewContainerNativeComponent.d.ts
+++ b/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewContainerNativeComponent.d.ts
@@ -1,10 +1,11 @@
-import { type ViewProps, type CodegenTypes } from 'react-native';
+import { type ViewProps } from 'react-native';
+import type { Float } from 'react-native/Libraries/Types/CodegenTypes';
 export interface NativeProps extends ViewProps {
     /**
      * The distance between elements at which they begin to merge.
      * Defaults to 0.
      */
-    spacing?: CodegenTypes.Float;
+    spacing?: Float;
 }
 declare const _default: import("react-native/types_generated/Libraries/Utilities/codegenNativeComponent").NativeComponentType<NativeProps>;
 export default _default;
diff --git a/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewNativeComponent.d.ts b/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewNativeComponent.d.ts
index 5a27de6..922da4d 100644
--- a/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewNativeComponent.d.ts
+++ b/node_modules/@callstack/liquid-glass/lib/typescript/src/LiquidGlassViewNativeComponent.d.ts
@@ -1,12 +1,13 @@
-import { type ViewProps, type ColorValue, type CodegenTypes } from 'react-native';
+import { type ViewProps, type ColorValue } from 'react-native';
+import type { Float, WithDefault } from 'react-native/Libraries/Types/CodegenTypes';
 export interface NativeProps extends ViewProps {
     interactive?: boolean;
-    effect?: CodegenTypes.WithDefault<'clear' | 'regular' | 'none', 'regular'>;
-    animated?: CodegenTypes.WithDefault<boolean, true>;
-    animationDuration?: CodegenTypes.WithDefault<CodegenTypes.Float, 0>;
-    hasAnimationDuration?: CodegenTypes.WithDefault<boolean, false>;
+    effect?: WithDefault<'clear' | 'regular' | 'none', 'regular'>;
+    animated?: WithDefault<boolean, true>;
+    animationDuration?: WithDefault<Float, 0>;
+    hasAnimationDuration?: WithDefault<boolean, false>;
     tintColor?: ColorValue;
-    colorScheme?: CodegenTypes.WithDefault<'light' | 'dark' | 'system', 'system'>;
+    colorScheme?: WithDefault<'light' | 'dark' | 'system', 'system'>;
 }
 declare const _default: import("react-native/types_generated/Libraries/Utilities/codegenNativeComponent").NativeComponentType<NativeProps>;
 export default _default;
diff --git a/node_modules/@callstack/liquid-glass/src/LiquidGlassViewContainerNativeComponent.ts b/node_modules/@callstack/liquid-glass/src/LiquidGlassViewContainerNativeComponent.ts
index 40dae13..f074a4b 100644
--- a/node_modules/@callstack/liquid-glass/src/LiquidGlassViewContainerNativeComponent.ts
+++ b/node_modules/@callstack/liquid-glass/src/LiquidGlassViewContainerNativeComponent.ts
@@ -1,15 +1,15 @@
 import {
   codegenNativeComponent,
   type ViewProps,
-  type CodegenTypes,
 } from 'react-native';
+import type { Float } from 'react-native/Libraries/Types/CodegenTypes';
 
 export interface NativeProps extends ViewProps {
   /**
    * The distance between elements at which they begin to merge.
    * Defaults to 0.
    */
-  spacing?: CodegenTypes.Float;
+  spacing?: Float;
 }
 
 export default codegenNativeComponent<NativeProps>('LiquidGlassContainerView');
diff --git a/node_modules/@callstack/liquid-glass/src/LiquidGlassViewNativeComponent.ts b/node_modules/@callstack/liquid-glass/src/LiquidGlassViewNativeComponent.ts
index 4e87aae..fa4e8bc 100644
--- a/node_modules/@callstack/liquid-glass/src/LiquidGlassViewNativeComponent.ts
+++ b/node_modules/@callstack/liquid-glass/src/LiquidGlassViewNativeComponent.ts
@@ -2,17 +2,17 @@ import {
   codegenNativeComponent,
   type ViewProps,
   type ColorValue,
-  type CodegenTypes,
 } from 'react-native';
+import type { Float, WithDefault } from 'react-native/Libraries/Types/CodegenTypes';
 
 export interface NativeProps extends ViewProps {
   interactive?: boolean;
-  effect?: CodegenTypes.WithDefault<'clear' | 'regular' | 'none', 'regular'>;
-  animated?: CodegenTypes.WithDefault<boolean, true>;
-  animationDuration?: CodegenTypes.WithDefault<CodegenTypes.Float, 0>;
-  hasAnimationDuration?: CodegenTypes.WithDefault<boolean, false>;
+  effect?: WithDefault<'clear' | 'regular' | 'none', 'regular'>;
+  animated?: WithDefault<boolean, true>;
+  animationDuration?: WithDefault<Float, 0>;
+  hasAnimationDuration?: WithDefault<boolean, false>;
   tintColor?: ColorValue;
-  colorScheme?: CodegenTypes.WithDefault<'light' | 'dark' | 'system', 'system'>;
+  colorScheme?: WithDefault<'light' | 'dark' | 'system', 'system'>;
 }
 
 export default codegenNativeComponent<NativeProps>('LiquidGlassView');
