diff --git a/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm b/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm
index f1d55dc..fdd59d2 100644
--- a/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm
+++ b/node_modules/@callstack/liquid-glass/ios/LiquidGlassContainerView.mm
@@ -42,29 +42,33 @@ using namespace facebook::react;
   return self;
 }
 
-#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
-    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
-
 - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
 {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   const auto &oldViewProps = *std::static_pointer_cast<LiquidGlassContainerViewProps const>(_props);
   const auto &newViewProps = *std::static_pointer_cast<LiquidGlassContainerViewProps const>(props);
   
   if (oldViewProps.spacing != newViewProps.spacing) {
     [_view setSpacing:newViewProps.spacing];
   }
+#endif
 
   [super updateProps:props oldProps:oldProps];
 }
 
 - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   [_view.contentView insertSubview:childComponentView atIndex:index];
+#else
+  [_view insertSubview:childComponentView atIndex:index];
+#endif
 }
 
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
   [childComponentView removeFromSuperview];
 }
-#endif
 
 @end
 
diff --git a/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm b/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm
index 2cfc99b..e59f43b 100644
--- a/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm
+++ b/node_modules/@callstack/liquid-glass/ios/LiquidGlassView.mm
@@ -45,16 +45,19 @@ using namespace facebook::react;
   return self;
 }
 
-#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
-    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
 - (void)layoutSubviews {
   [super layoutSubviews];
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   _view.layer.cornerRadius = self.layer.cornerRadius;
   _view.layer.cornerCurve = self.layer.cornerCurve;
+#endif
 }
 
 - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
 {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   const auto &oldViewProps = *std::static_pointer_cast<LiquidGlassViewProps const>(_props);
   const auto &newViewProps = *std::static_pointer_cast<LiquidGlassViewProps const>(props);
   BOOL needsSetup = NO;
@@ -125,6 +128,7 @@ using namespace facebook::react;
   if (needsSetup) {
     [_view setupView];
   }
+#endif
   
   [super updateProps:props oldProps:oldProps];
 }
@@ -138,6 +142,8 @@ using namespace facebook::react;
   
   _needsInvalidateLayer = NO;
   
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   if (@available(iOS 26.0, tvOS 26.0, *)) {
     const auto borderMetrics = _props->resolveBorderMetrics(_layoutMetrics);
 
@@ -156,6 +162,7 @@ using namespace facebook::react;
 
     _view.cornerConfiguration = [UICornerConfiguration configurationWithTopLeftRadius:topLeft topRightRadius:topRight bottomLeftRadius:bottomLeft bottomRightRadius:bottomRight];
   }
+#endif
 }
 
 - (void)updateLayoutMetrics:(const LayoutMetrics &)layoutMetrics
@@ -170,13 +177,16 @@ using namespace facebook::react;
 }
 
 - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 260000) || \
+    (defined(__TV_OS_VERSION_MAX_ALLOWED) && __TV_OS_VERSION_MAX_ALLOWED >= 260000)
   [_view.contentView insertSubview:childComponentView atIndex:index];
+#else
+  [_view insertSubview:childComponentView atIndex:index];
+#endif
 }
 
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index {
   [childComponentView removeFromSuperview];
 }
 
-#endif
-
 @end
